### ISSUE TITLE
vkconfig: Enabled dev sim layer and the monitor layer, then moving the layer orders caused the application to crash

### ISSUE THREAD

**Issue by johnzupin** [LunarG/VulkanTools#1449]
Environment:
Ubuntu 20.04
Nvidia

Steps to reproduce:
1. Right click and select new
2. Select Overriden/ForceLayer the dev sim layer and click ok to the popup window
3. Then select Overridden/Force Layer to the monitor layer
4. Select the monitor layer in the order window below, move the monitor layer down toward the driver (app should crash)

**Comment by johnzupin** [comment]
![dev-monitor-crash](https://user-images.githubusercontent.com/3836693/115422183-e3ee5080-a1b9-11eb-8347-c6b5ade1b30f.gif)

**Comment by johnzupin** [comment]
```
tarting program: /home/lunarg/VulkanTools/dbuild/vkconfig/vkconfig 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff2d9f700 (LWP 1590)]
[New Thread 0x7ffff1204700 (LWP 1591)]
[New Thread 0x7ffff0a03700 (LWP 1592)]
[New Thread 0x7fffe3efe700 (LWP 1593)]
UNASSIGNED-khronos-validation-createinstance-status-message(INFO / SPEC): msgNum: -671457468 - Validation Information: [ UNASSIGNED-khronos-validation-createinstance-status-message ] Object 0: handle = 0x555555b47ac0, type = VK_OBJECT_TYPE_INSTANCE; | MessageID = 0xd7fa5f44 | Khronos Validation Layer Active:
    Settings File: Found at /home/lunarg/.local/share/vulkan/settings.d/vk_layer_settings.txt specified by VkConfig application override.
    Current Enables: None.
    Current Disables: VK_VALIDATION_FEATURE_DISABLE_THREAD_SAFETY_EXT.

    Objects: 1
        [0] 0x555555b47ac0, type: 1, name: NULL
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc

Thread 1 "vkconfig" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff6705859 in __GI_abort () at abort.c:79
#2  0x00007ffff6adc951 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff6ae847c in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff6ae84e7 in std::terminate() ()
   from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff6ae8799 in __cxa_throw ()
   from /lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007ffff6adc562 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00007ffff6b80dbf in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_assign(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
   from /lib/x86_64-linux-gnu/libstdc++.so.6
#8  0x00007ffff6b8112e in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::operator=(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
   from /lib/x86_64-linux-gnu/libstdc++.so.6
#9  0x000055555558c3f8 in LayersDialog::OverrideOrder (this=0x7fffffffce30, 
    layer_name=<error: Cannot access memory at address 0x73005f00650063>, 
    below=0x5555565ea150, above=0x5555565ea080)
    at /home/lunarg/VulkanTools/vkconfig/dialog_layers.cpp:351
#10 0x000055555558c66b in LayersDialog::on_pushButtonDown_clicked (
    this=0x7fffffffce30)
--Type <RET> for more, q to quit, c to continue without paging-- 
    at /home/lunarg/VulkanTools/vkconfig/dialog_layers.cpp:372
#11 0x000055555557c3a5 in LayersDialog::qt_static_metacall (_o=0x7fffffffce30, 
    _c=QMetaObject::InvokeMetaMethod, _id=5, _a=0x7fffffffc000)
    at /home/lunarg/VulkanTools/dbuild/vkconfig/vkconfig_autogen/EWIEGA46WW/moc_dialog_layers.cpp:110
#12 0x000055555557c504 in LayersDialog::qt_metacall (this=0x7fffffffce30, 
    _c=QMetaObject::InvokeMetaMethod, _id=5, _a=0x7fffffffc000)
    at /home/lunarg/VulkanTools/dbuild/vkconfig/vkconfig_autogen/EWIEGA46WW/moc_dialog_layers.cpp:148
#13 0x00007ffff6ed13bc in QMetaObject::activate(QObject*, int, int, void**) ()
   from /lib/x86_64-linux-gnu/libQt5Core.so.5
#14 0x00007ffff7b73806 in QAbstractButton::clicked(bool) ()
   from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#15 0x00007ffff7b73a2e in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#16 0x00007ffff7b74e73 in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#17 0x00007ffff7b75035 in QAbstractButton::mouseReleaseEvent(QMouseEvent*) ()
   from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#18 0x00007ffff7ac12b6 in QWidget::event(QEvent*) ()
   from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#19 0x00007ffff7a7ea66 in QApplicationPrivate::notify_helper(QObject*, QEvent*)
    () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#20 0x00007ffff7a88343 in QApplication::notify(QObject*, QEvent*) ()
   from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
--Type <RET> for more, q to quit, c to continue without paging-- 
#21 0x00007ffff6ea593a in QCoreApplication::notifyInternal2(QObject*, QEvent*)
    () from /lib/x86_64-linux-gnu/libQt5Core.so.5
```

**Comment by johnzupin** [comment]
Still seeing this issue as of ref: v1.2.170-261-g5aeab5532

**Comment by christophe-lunarg** [comment]
I could reproduce this bug, I'll fix it tomorrow

**Comment by christophe-lunarg** [comment]
This issue is resolved by https://github.com/LunarG/VulkanTools/pull/1456, great catch!

**Comment by johnzupin** [comment]
Confirming that this is fixed as of ref: v1.2.176
